token/program-2022-test/tests/token_group_initialize_member.rs

```diff
@@ -0,0 +1,587 @@
+    let fetched_member = pod_from_bytes::<TokenGroupMember>(member_bytes).unwrap();
+    assert_eq!(
+        fetched_member,
+        &TokenGroupMember {
+            mint: member1_mint_keypair.pubkey(),
+            group: group_mint_keypair.pubkey(),
+            member_number: 1.try_into().unwrap(),
+        }
+    );
+
+    // fail double-init
+    let error = member1_token_context
+        .token
+        .token_group_initialize_member(
+            &member1_token_context.mint_authority.pubkey(),
+            &group_mint_keypair.pubkey(),
+            &group_authority.pubkey(),
+            &[&member1_token_context.mint_authority, &group_authority],
+        )
+        .await
+        .unwrap_err();
+    assert_eq!(
+        error,
+        TokenClientError::Client(Box::new(TransportError::TransactionError(
+            TransactionError::InstructionError(
+                0,
+                InstructionError::Custom(TokenError::ExtensionAlreadyInitialized as u32)
+            )
+        )))
+    );
+
+    // Now the others
+    member2_token_context
+        .token
+        .token_group_initialize_member_with_rent_transfer(
+            &payer.pubkey(),
+            &member2_token_context.mint_authority.pubkey(),
+            &group_mint_keypair.pubkey(),
+            &group_authority.pubkey(),
+            &[
+                &payer,
+                &member2_token_context.mint_authority,
+                &group_authority,
+            ],
+        )
+        .await
+        .unwrap();
+    let mint_info = member2_token_context.token.get_mint_info().await.unwrap();
+    let member_bytes = mint_info.get_extension_bytes::<TokenGroupMember>().unwrap();
+    let fetched_member = pod_from_bytes::<TokenGroupMember>(member_bytes).unwrap();
+    assert_eq!(
+        fetched_member,
+        &TokenGroupMember {
+            mint: member2_mint_keypair.pubkey(),
+            group: group_mint_keypair.pubkey(),
+            member_number: 2.try_into().unwrap(),
+        }
+    );
+
+    member3_token_context
+        .token
+        .token_group_initialize_member_with_rent_transfer(
+            &payer.pubkey(),
+            &member3_token_context.mint_authority.pubkey(),
+            &group_mint_keypair.pubkey(),
+            &group_authority.pubkey(),
+            &[
+                &payer,
+                &member3_token_context.mint_authority,
+                &group_authority,
+            ],
+        )
+        .await
+        .unwrap();
+    let mint_info = member3_token_context.token.get_mint_info().await.unwrap();
+    let member_bytes = mint_info.get_extension_bytes::<TokenGroupMember>().unwrap();
+    let fetched_member = pod_from_bytes::<TokenGroupMember>(member_bytes).unwrap();
+    assert_eq!(
+        fetched_member,
+        &TokenGroupMember {
+            mint: member3_mint_keypair.pubkey(),
+            group: group_mint_keypair.pubkey(),
+            member_number: 3.try_into().unwrap(),
+        }
+    );
```

Comment:
It would exercise more code if the test sets a max size of 2, which makes this fail. What do you think?